src/libsyntax_ext/proc_macro_registrar.rs

```diff
@@ -286,25 +350,36 @@ fn mk_registrar(cx: &mut ExtCtxt,
     let registry = Ident::from_str("Registry");
     let registrar = Ident::from_str("registrar");
     let register_custom_derive = Ident::from_str("register_custom_derive");
-    let stmts = custom_derives.iter().map(|cd| {
+    let register_custom_attribute = Ident::from_str("register_custom_attribute");
+
+    let mut stmts = custom_derives.iter().map(|cd| {
         let path = cx.path_global(cd.span, vec![cd.function_name]);
         let trait_name = cx.expr_str(cd.span, cd.trait_name);
         let attrs = cx.expr_vec_slice(
-            span,
+            cd.span,
             cd.attrs.iter().map(|&s| cx.expr_str(cd.span, s)).collect::<Vec<_>>()
         );
-        (path, trait_name, attrs)
-    }).map(|(path, trait_name, attrs)| {
-        let registrar = cx.expr_ident(span, registrar);
-        let ufcs_path = cx.path(span, vec![proc_macro, __internal, registry,
+        let registrar = cx.expr_ident(cd.span, registrar);
+        let ufcs_path = cx.path(cd.span, vec![proc_macro, __internal, registry,
                                            register_custom_derive]);
-        cx.expr_call(span,
-                     cx.expr_path(ufcs_path),
-                     vec![registrar, trait_name, cx.expr_path(path), attrs])
-    }).map(|expr| {
-        cx.stmt_expr(expr)
+
+        cx.stmt_expr(cx.expr_call(cd.span, cx.expr_path(ufcs_path),
+                                    vec![registrar, trait_name, cx.expr_path(path), attrs]))
```

Comment:
If the line break is here, it should be indented up to the second `(`.
I would write this:
```rust
        let args = vec![registrar, trait_name, cx.expr_path(path), attrs];
        cx.stmt_expr(cx.expr_call(cd.span, cx.expr_path(ufcs_path), args));
```